spec/models/manageiq/providers/amazon/regions_spec.rb

```diff
@@ -1,18 +1,22 @@
 describe ManageIQ::Providers::Amazon::Regions do
+  # https://github.com/aws/aws-sdk-ruby/blob/5fe5795e8910bb667996dfc75e4f16b7e69e3980/gems/aws-partitions/partitions.json#L11
+  let(:default_regions_regexp) { /^(us|eu|ap|sa|ca)\-\w+\-\d+$/ }
+
   it "has all the regions" do
     ems = FactoryGirl.create(:ems_amazon_with_vcr_authentication)
 
     VCR.use_cassette(described_class.name.underscore) do
       current_regions = described_class.regions.map do |_name, config|
         {:region_name => config[:name], :endpoint => config[:hostname]}
       end
-      current_regions.reject! { |r| r[:region_name] == 'us-gov-west-1' }
+      current_regions.select! { |r| r[:region_name] =~ default_regions_regexp }
```

Comment:
How does this and the regex above handle gov cloud? And potentially other regions like this PR https://github.com/aws/aws-sdk-ruby/blob/5fe5795e8910bb667996dfc75e4f16b7e69e3980/gems/aws-partitions/partitions.json#L11 aka China.

I'm not sure I'm understanding the regex.

Reply:
I took that regexp from that file as it is. It's something from AWS internals.

You can find the China's regexp here: https://github.com/aws/aws-sdk-ruby/blob/5fe5795e8910bb667996dfc75e4f16b7e69e3980/gems/aws-partitions/partitions.json#L2058
And GovCloud's here: https://github.com/aws/aws-sdk-ruby/blob/5fe5795e8910bb667996dfc75e4f16b7e69e3980/gems/aws-partitions/partitions.json#L2359

This spec works the same way as before - it is hard to test those partitions (in the AWS gem's terminology) simply because our credentials designated for default partition only, and AWS's `DescribeRegions` call (no matter, will it be gem, some other library, or cli) returns only those, whose available for our account).

Reply:
> and AWS's DescribeRegions call (no matter, will it be gem, some other library, or cli) returns only those, whose available for our account).

Doesn't this potentially cause us to miss regions if our account doesn't see them?

Reply:
>  I took that regexp from that file as it is. It's something from AWS internals.

This regex ignores a bunch of regions, but the purpose of this test it to verify we have *all* regions, so I'm not sure the purposes match.

Reply:
Hi @Fryguy, we are not able to get other regions from AWS, with any tool. It is limited by our AWS account. As you can see, previous test version was doing the same.

Reply:
P.S. I even don't see any advantage in this test. It just tests one recording against another.

Reply:
@AlexanderZagaynov Yeah, I think I came to the same conclusion that the test is really a nonsense test.  I do wonder if we can create a test that shows we have *all* possible regions.

Reply:
@Fryguy since my script takes all possible regions from AWS gem's repo source, I see no point at all to have a such test (it will match anyway). If it will be some VCR recording, we will be simply comparing two recordings (one from VCR, another from that `yml` file). On each regions update both recordings should be updated.
Even if we will have proper AWS credentials to check China and Gov clouds - we will stay with the same VCR recordings.
So the only point to have that test - to check if `yml` file have proper format?
If your goal to check do we have _all_ possible regions, considering that we can have AWS's regions list change unnoticed - I would suggest to check that manually during QA stage. One should open Amazon's official regions list (from their docs site), and compare with what we have. Even if it will be automated - VCR records will make no sense, need to have a fresh info on each test run.

Reply:
@AlexanderZagaynov I agree.  I'll merge as is for now, and we can choose to drop this spec in a separate PR.